### PR TITLE
Feature/pod defaults

### PR DESCRIPTION
**Pull request by jaystary**
- Add pod-default module
- Updated Cognito module by removing static phone number

### CODE REVIEW

**Comment by ncbkr** [comment]
@jaystary can you please elaborate a bit more on the Pod defaults and explain what's the purpose of this new feature?

File: modules/kubeflow-pod-default/variables.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  description = "The ARN of the role attached to the external-secret deployment. This is the role that will by default be assumed if roleArn is not specified in the ExternalSecret kubernetes spec"
+}
+
+variable "external_secrets_secret_role_arn" {
+  type  = string
+  default = ""
+  description = "The ARN of the role that should be assumed by the external-secret deployment when creating the MLFlow ExternalSecret. This role must be assumable by the role that has been attached to external-secret deployment's service account. If left blank, a role will be created."
+}
+
+/*
```
Comment (ncbkr) [inline modules/kubeflow-pod-default/variables.tf:12]: What is the purpose of those commented out variables? Are we planning on adding them later? If not,  I would remove them.

**Review by ncbkr — COMMENTED** [review]
Looks fine to me. I would like to see a more detailed description of the changes of the pull request and their effects. Also some brieve comment on how to use pod defaults.

File: modules/kubeflow-pod-default/variables.tf (hunk 1 above)
Comment (jaystary) [reply modules/kubeflow-pod-default/variables.tf:12]: Commented out can be removed

**Comment by jaystary** [comment]
> @jaystary can you please elaborate a bit more on the Pod defaults and explain what's the purpose of this new feature?

PodDefault alas admission-webhook (https://github.com/kubeflow/kubeflow/blob/master/components/admission-webhook/README.md) allows you to select and write preset variables into your Kubeflow notebook.

The workflow we implemented is, we pull from an external secrets manager and write them into our cluster:
https://github.com/external-secrets/kubernetes-external-secrets

**Pull request merged**